airbyte-integrations/connectors/source-shopify/source_shopify/streams/streams.py

```diff
@@ -327,5 +332,88 @@ class CustomerAddress(IncrementalShopifyGraphQlBulkStream):
+        # only one top page in query
+        delivery_profiles_nodes = json_response.get("deliveryProfiles", {}).get("nodes")
+        if delivery_profiles_nodes:
+            profile_location_groups = delivery_profiles_nodes[0].get("profileLocationGroups")
+            if profile_location_groups:
+                sub_page_info = (
+                    # only first one
+                    profile_location_groups[0].get("locationGroupZones", {}).get("pageInfo", {})
+                )
+
+        if not sub_page_info["hasNextPage"] and not page_info["hasNextPage"]:
+            return None
+        if sub_page_info["hasNextPage"]:
+            # The cursor to retrieve nodes after in the connection. Typically, you should pass the endCursor of the previous page as after.
+            self._sub_page_cursor = sub_page_info["endCursor"]
+        if page_info["hasNextPage"] and not sub_page_info["hasNextPage"]:
+            # The cursor to retrieve nodes after in the connection. Typically, you should pass the endCursor of the previous page as after.
+            self._page_cursor = page_info["endCursor"]
+            self._sub_page_cursor = None
+
+        return {
+            "cursor": self._page_cursor,
+            "sub_cursor": self._sub_page_cursor,
+        }
+
+    def request_body_json(
+        self,
+        stream_state: Optional[Mapping[str, Any]],
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping[str, Any]]:
+        location_group_id = stream_slice["parent"]["profile_location_groups"][0]["locationGroup"]["id"]
+        return {
+            "query": self.query(location_group_id=location_group_id, location_group_zones_cursor=self._sub_page_cursor).get(
+                query_args={
+                    "cursor": self._page_cursor,
+                }
+            ),
+        }
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for node in super().parse_response(response, **kwargs):
+            for location_group in node.get("profileLocationGroups", []):
+                for location_group_zone in location_group.get("locationGroupZones", {}).get("nodes", []):
+                    for country in location_group_zone.get("zone", {}).get("countries"):
+                        country = self._process_country(country)
+                        if country["id"] not in self._synced_countries_ids:
+                            self._synced_countries_ids.append(country["id"])
+                            yield country
+
+    def _process_country(self, country: Mapping[str, Any]) -> Mapping[str, Any]:
+        country["id"] = int(country["id"].split("/")[-1])
+
+        for province in country.get("provinces", []):
+            province["id"] = int(province["id"].split("/")[-1])
+            province["country_id"] = country["id"]
+
+        country["rest_of_world"] = country["code"]["rest_of_world"]
```

Comment:
Let's make sure we have a default values to fallback to, if, for instance: `country["code"]["rest_of_world"]` is not available. Can we use `.get()` instead? Are we 100% sure the values would be available?

Reply:
I'm 99% sure that those fields will be in a response, but added check + default values for safety